angularSTEP/src/app/user-page/user-page.component.ts

```diff
@@ -55,4 +74,62 @@ export class UserPageComponent implements OnInit {
       }
     }
   }
+
+  async isUserFollowing() {
+    if (this.currentPageUid && this.loggedIn && this.currentUserUid) {
+      const postUser = await this.afs.firestore
+        .doc('/users/' + this.currentUserUid + '/')
+        .withConverter(new Converter().userConverter)
+        .get();
+      if (postUser) {
+        this.currentUserData = postUser.data();
+        if (
+          this.currentUserData &&
+          this.currentUserData.following.indexOf(this.currentPageUid) > -1
+        ) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  follow() {
```

Comment:
These follow() and unfollow() functions seem like good candidates for unit testing coverage

Reply:
Yes, I will be sure to make unit testing for them. Thank you!